dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchCollection.cs

```diff
@@ -125,53 +125,77 @@ public override async Task<bool> CollectionExistsAsync(CancellationToken cancell
     }
 
     /// <inheritdoc />
-    public override Task CreateCollectionAsync(CancellationToken cancellationToken = default)
+    public override async Task EnsureCollectionExistsAsync(CancellationToken cancellationToken = default)
     {
-        var vectorSearchConfig = new VectorSearch();
-        var searchFields = new List<SearchField>();
+        const string OperationName = "CreateIndex";
 
-        // Loop through all properties and create the search fields.
-        foreach (var property in this._model.Properties)
+        // Don't even try to create if the collection already exists.
+        if (await this.CollectionExistsAsync(cancellationToken).ConfigureAwait(false))
         {
-            switch (property)
+            return;
+        }
+
+        try
+        {
+            var vectorSearchConfig = new VectorSearch();
```

Comment:
Can most of this be extacted out of the `try`, so that the `try` scope is as small as possible?